From c9c6b2f20eeff9e693988657d6add5075c1930f0 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Thu, 4 Apr 2024 18:14:33 +0200
Subject: [PATCH 0763/1695] dt-bindings: mfd: describe STM32MP25 PWR

Describe STM32 MP25 PWR block as an MFD device
as it handles regulators and interrupts.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git ba6ba5c639e172b3ad9f8f19b5d8ca1c874f13e1 dt-bindings: mfd: describe STM32MP25 PWR]
Change-Id: I3e0d81c802818d408dfce43bfed35ae83fb45ef6
Signed-off-by: Pascal Paillet <p.paillet@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/371498
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/373304
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 .../bindings/mfd/st,stm32mp25-pwr.yaml        | 95 +++++++++++++++++++
 1 file changed, 95 insertions(+)
 create mode 100644 documentation/devicetree/bindings/mfd/st,stm32mp25-pwr.yaml

diff --git a/documentation/devicetree/bindings/mfd/st,stm32mp25-pwr.yaml b/documentation/devicetree/bindings/mfd/st,stm32mp25-pwr.yaml
new file mode 100644
index 000000000..3af8401f3
--- /dev/null
+++ b/documentation/devicetree/bindings/mfd/st,stm32mp25-pwr.yaml
@@ -0,0 +1,95 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/mfd/st,stm32mp25-pwr.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STM32MP25 PWR
+
+description: STMicroelectronics STM32MP25 PWR peripheral
+  This hardware block provides some power supplies
+  configuration and wake-up pin interrupts.
+
+maintainers:
+  - Pascal Paillet <p.paillet@st.com>
+
+properties:
+  compatible:
+    const: st,stm32mp25-pwr
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    description: contains the PWR Wake-up pin interrupt line.
+    maxItems: 1
+
+  wakeup-parent: true
+
+  wakeup-gpios:
+    description: contains a list of GPIO spec describing each wake-up pin.
+    maxItems: 6
+
+patternProperties:
+  "^(vdd33ucpd|vdda18adc|vddgpu|vddio|vddio1|vddio2|vddio3|vddio4)$":
+    type: object
+
+    $ref: "regulator.yaml#"
+
+    properties:
+      compatible:
+        const: st,stm32mp25-pwr-regu
+
+      vdd-supply:
+        description: Input supply phandle(s) for vdd and vddio input
+
+    patternProperties:
+      ".*-supply$":
+        description: Input supply phandle(s) for this node
+
+required:
+  - compatible
+  - reg
+
+additionalProperties: false
+
+examples:
+  - |
+    pwr: pwr@44210000 {
+      compatible = "st,stm32mp25-pwr";
+      reg = <0x44210000 0x400>;
+      wakeup-parent = <&exti1>;
+      wakeup-gpios = <0>, <0>, <0>, <0>, <&gpiog 2 GPIO_PULL_UP>, <0>;
+      interrupts = <GIC_SPI 226 IRQ_TYPE_LEVEL_HIGH>;
+
+      vdd33ucpd: vdd33ucpd {
+        compatible = "st,stm32mp25-pwr-regu";
+        regulator-name = "vdd33ucpd";
+        vdd33ucpd-supply = <&vdd3v3_usb>;
+      };
+
+      vdda18adc: vdda18adc {
+        compatible = "st,stm32mp25-pwr-regu";
+        regulator-name = "vdda18adc";
+        vdda18adc-supply = <&v1v8>;
+      };
+
+      vddgpu: vddgpu {
+        compatible = "st,stm32mp25-pwr-regu";
+        regulator-name = "vddgpu";
+        vddgpu-supply = <&vddgpu_pmic>;
+      };
+
+      vddio1: vddio1 {
+        compatible = "st,stm32mp25-pwr-regu";
+        regulator-name = "vddio1";
+        vddio1-supply = <&vddio_sdcard>;
+      };
+
+      vddio: vddio {
+        compatible = "st,stm32mp25-pwr-regu";
+        regulator-name = "vddio";
+        vdd-supply = <&vdd>;
+      };
+    };
+...
-- 
2.47.2

